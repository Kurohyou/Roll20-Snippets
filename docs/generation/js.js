const fs = require(`fs/promises`);
const hb = require(`handlebars`);
const markdown = require(`markdown`).markdown;

const parseJSdoc = async (docData,docPath) => {
  const template = await fs.readFile(`${docPath}/templates/js.handlebars`,`utf8`)
    .then(t => hb.compile(t));
  // console.log(`TEMPLATE`,template);
  docData.sort((a,b)=>a.name.localeCompare(b.name));
  const rendered = template({docData});
  return fs.writeFile(`${docPath}/js.html`,rendered);
};

module.exports = parseJSdoc;