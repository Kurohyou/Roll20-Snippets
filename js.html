<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <link rel="preconnect" href="https://fonts.googleapis.com"/>
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="crossorigin"/>
    <link href="https://fonts.googleapis.com/css2?family=Aldrich&amp;family=Exo+2:ital,wght@0,400;0,700;1,400&amp;family=Fira+Code&amp;display=swap" rel="stylesheet"/>
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"/>
    <link rel="stylesheet" href="assets/portfolio/social-icons.css"/>
    <link rel="stylesheet" href="assets/css/clear.css"/>
    <link rel="stylesheet" href="assets/css/generic.css"/>
    <link rel="stylesheet" href="assets/css/prettify.css"/>
    <link rel="stylesheet" href="assets/css/sons-of-obsidian.css"/>
    <link rel="stylesheet" href="assets/css/codeDisplay.css"/>
    <title>K-Scaffold | Sheetworker</title>
  </head>
  <body>
    <header>
      <h1>The K-Scaffold Framework</h1><span class="subhead">PUG, HTML, and Javascript</span>
    </header>
    <nav>
      <ul>
        <li><a href="index.html?v=1.1">General Info</a></li>
        <li><a href="pug.html?v=1.1">Pug Documentation</a></li>
      </ul>
      <ul>
        <li><a href="#intro-header">Intro</a></li>
        <li><a href="#" id="doc-link">K-Scaffold methods and variables</a></li>
        <li><a href="#contact-header">Contact & Feedback</a></li>
      </ul>
    </nav>
    <main id="doc-target">
      <section aria-labelledby="intro-header">
        <h2 id="intro-header">js library</h2>
        <div class="border"></div>
        <div class="content">
          <p>The K-Scaffold's javascript library provides a wide range of functionality. The library's functions are divided into three broad types; aliases for Roll20 sheetworker functions, utility functions, and K-Scaffold specific functions.</p>
          <h3>Sheetworker Aliases</h3>
          <p>The K-Scaffold's sheetworker aliases provide improved versions of many of Roll20's sheetworker functions. These aliases provide combinations of sheetworker functions that are frequently used together.</p>
          <h3>Utility functions</h3>
          <p>The scaffold's utility functions provide many general use functions that can be used to do everything from parse a repeating attribute name into its component parts to querying the user about what they want to do.</p>
          <h3>K-Scaffold specific functions</h3>
          <p>The K-scaffold specific functions provide methods for hooking your functions into the K-scaffold's functionality.</p>
        </div>
      </section>
    </main>
    <aside>
      <h2>Table of Contents</h2>
      <nav>
        <ul id="toc-target"></ul>
      </nav>
    </aside>
    <div class="footer"></div>
    <footer>
      <section class="flex flex--space-evenly" aria-label="Contact &amp; Feedback" id="contact-header"><a class="social-icon" href="https://discord.gg/vYmvFBZQsM">discord</a><a class="social-icon" href="https://www.linkedin.com/in/scott-casey-20210398/">linkedin</a><a class="social-icon" href="https://www.patreon.com/kurohyoustudios?fan_landing=true">patreon</a>
        <div class="flex flex-column"><span style="justify-self:center;">Scott Casey | Kurohyou Studios</span><span style="justify-self:center;">06/19/2022</span></div><a class="material-icons" title="Email" href="mailto:scaseydv@gmail.com?subject=Inquiring about your work">email</a><a class="social-icon" href="https://github.com/Kurohyou">github</a><a class="social-icon" href="https://twitter.com/Kurohyoustudios">twitter</a>
      </section>
    </footer>
    <script type="text/javascript">const siteVersion = 1.1;
      
    </script>
    <script src="assets/javascript/resizer.js"></script>
    <script src="assets/javascript/prettify.js"></script>
    <script src="assets/javascript/parser.js"></script>
  </body>
</html>