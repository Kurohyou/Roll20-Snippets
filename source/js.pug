include _mixins.pug
doctype html
html(lang="en")
  head
    meta(charset="UTF-8")
    meta(http-equiv="X-UA-Compatible", content="IE=edge")
    +generalStyleSheets('js')
    +codeSheetStyles
    title K-Scaffold | Sheetworker
  body
    +pageHead('js')
      li
        a(href="#intro-header") Intro
      li
        a(href="#")#doc-link K-Scaffold methods and variables
    main#doc-target
      +mainSection('js library','intro-header')
        p The K-Scaffold's javascript library provides a wide range of functionality. The library's functions are divided into three broad types; aliases for Roll20 sheetworker functions, utility functions, and K-Scaffold specific functions.
        h3 Sheetworker Aliases
        p The K-Scaffold's sheetworker aliases provide improved versions of many of Roll20's sheetworker functions. These aliases provide combinations of sheetworker functions that are frequently used together.
        h3 Utility functions
        p The scaffold's utility functions provide many general use functions that can be used to do everything from parse a repeating attribute name into its component parts to querying the user about what they want to do.
        h3 K-Scaffold specific functions
        p The K-scaffold specific functions provide methods for hooking your functions into the K-scaffold's functionality.
    aside
      h2 Table of Contents
      nav
        ul#toc-target
    +pageFooter
    +codePageScripts