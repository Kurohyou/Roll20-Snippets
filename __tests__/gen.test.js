import { describe, expect, it } from 'vitest';

import { environment } from '../test_sheet/source/assets/js/testFramework';

describe('Environment',() => {
  it(
    'Should have attributes, characters, query response sub objects and all attributes from sheet in attributes subobject',
    ()=>{
      expect(environment).toMatchSnapshot();
    }
  );
});