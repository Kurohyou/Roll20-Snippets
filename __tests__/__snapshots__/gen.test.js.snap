// Vitest Snapshot v1

exports[`Environment > Should have attributes, characters, query response sub objects and all attributes from sheet in attributes subobject 1`] = `
{
  "attributes": {
    "action_text": "",
    "adaptive_input": "",
    "adaptive_textarea": "",
    "basic_collapse": 0,
    "basic_input": "test",
    "button_text": "",
    "checked": "1",
    "drop_data": "",
    "drop_name": "",
    "fill_left": "0",
    "fill_left_with_display": "0",
    "hidden": "hidden",
    "image": "",
    "labelled_input": "",
    "labelled_roller_action": "",
    "labelled_select": "option 1",
    "main_tabs_tab": "nav-tabs-main-tabs--tab1",
    "number": "1",
    "prefix_drop_data": "",
    "prefix_drop_name": "",
    "radio": "1",
    "range": "2",
    "roller_action": "",
    "roller_text": "",
    "select": "option 1",
    "span": "",
    "template_start": "&{template:test} {{character_name=@{character_name}}} {{character_id=@{character_id}}}",
    "text": "text",
    "textarea": "textarea",
    "unchecked": 0,
  },
  "otherCharacters": {},
  "queryResponses": {},
  "triggers": [
    {
      "func": [Function],
      "trigger": "sheet:opened",
    },
    {
      "func": [Function],
      "trigger": "change:character_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:nav-tabs-main-tabs--basic-attributes",
    },
    {
      "func": [Function],
      "trigger": "clicked:nav-tabs-main-tabs--repeating-attributes",
    },
    {
      "func": [Function],
      "trigger": "clicked:nav-tabs-main-tabs--empty-tab",
    },
    {
      "func": [Function],
      "trigger": "change:select",
    },
    {
      "func": [Function],
      "trigger": "change:drop_name",
    },
    {
      "func": [Function],
      "trigger": "change:prefix_drop_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:action-button",
    },
    {
      "func": [Function],
      "trigger": "clicked:roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:labelled_select",
    },
    {
      "func": [Function],
      "trigger": "clicked:labelled_action",
    },
    {
      "func": [Function],
      "trigger": "clicked:labelled-roller-action",
    },
    {
      "func": [Function],
      "trigger": "remove:repeating_basic-fieldset",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_basic-fieldset:select",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_basic-fieldset:drop_name",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_basic-fieldset:prefix_drop_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_basic-fieldset:action-button",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_basic-fieldset:roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_basic-fieldset:labelled_select",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_basic-fieldset:labelled_action",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_basic-fieldset:labelled-roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_class-fieldset:select",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_class-fieldset:drop_name",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_class-fieldset:prefix_drop_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_class-fieldset:action-button",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_class-fieldset:roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_class-fieldset:labelled_select",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_class-fieldset:labelled_action",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_class-fieldset:labelled-roller-action",
    },
    {
      "func": [Function],
      "trigger": "clicked:add-custom-fieldset",
    },
    {
      "func": [Function],
      "trigger": "clicked:edit-custom-fieldset",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_custom-fieldset:select",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_custom-fieldset:drop_name",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_custom-fieldset:prefix_drop_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_custom-fieldset:action-button",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_custom-fieldset:roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_custom-fieldset:labelled_select",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_custom-fieldset:labelled_action",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_custom-fieldset:labelled-roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_inline-fieldset:collapse",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_inline-fieldset:select",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_inline-fieldset:drop_name",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_inline-fieldset:prefix_drop_name",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_inline-fieldset:action-button",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_inline-fieldset:roller-action",
    },
    {
      "func": [Function],
      "trigger": "change:repeating_inline-fieldset:labelled_select",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_inline-fieldset:labelled_action",
    },
    {
      "func": [Function],
      "trigger": "clicked:repeating_inline-fieldset:labelled-roller-action",
    },
  ],
}
`;
